### DISCUSSION TITLE
Some question about use snapshot in windows VM

### DISCUSSION THREAD

**Discussion by maocili** [cloud-hypervisor/cloud-hypervisor#3881] [Q&A] [answered]
I want to implement version control between different images. In the using snapshot ,  we meet some question as the followed.
1. For snapshot VM, I use snapshot again. The new snapshot fail to connection SAC and connect RDP,but network is working (can be ping )
2. For a RAW build two snapshot A and B. boot they together, the network of one of A or B doesn't work （they have differ mac and ip ）
3.  following the previous ,I write data in B and shutdown it, the data will be write in RAW. At the same time, restore snapshot A , I can read it.
In the end, I want express my gratitude for your taking time from your busy schedule to response my question.

**Comment by rbradford** [comment, marked as answer]
Hi thanks for experimenting with Cloud Hypervisor. It sounds like you're using the same disk image for multiple VMs and running them at the same time. That will cause issues. Only the memory and configuration of the VM forms the snapshot not the contents of the disk. With the latest version of Cloud Hypervisor the config file is inside the snapshot and is human readable/editable. If you're careful you could modify the disk image in a snapshot to point to a copy of the disk file from the original VM.
